boards/arm/stm32g081b_eval/board.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2022 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/adc.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(board, LOG_LEVEL_DBG);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(usbc_port)
+
+#define VBUS DT_PATH(vbus)
+
+/* Common settings supported by most ADCs */
+
+static int16_t sample_buffer;
+
+static const uint32_t output_ohm = DT_PROP(VBUS, output_ohms);
+static const uint32_t full_ohm = DT_PROP(VBUS, full_ohms);
+
+/* Data of ADC io-channels specified in devicetree. */
```

Comment:
this code seems now to be a copy from the other board, should likely be in the sample (and sample can be made non-board specific)

Reply:
Yes, for these two boards, the ADC code is similar but other boards could read VBUS very differently.

Reply:
why not create a sample level module for boards that use ADC? it can be compiled based on a sample level option choice, e.g. `MYSAMPLE_MEASURE_ADC`, which can in turn be enabled by board .conf overlay.

Reply:
I agree with you and refactored the code. I refactored the VBUS measurement and control into a device driver.